internal/metastore/kv/rootcoord/kv_catalog.go

```diff
@@ -1009,85 +1015,76 @@ func (kc *Catalog) AlterGrant(ctx context.Context, tenant string, entity *milvus
 		if !common.IsKeyNotExistError(err) {
 			return err
 		}
-		curGrantPrivilegeEntity.Entities = append(curGrantPrivilegeEntity.Entities, &milvuspb.GrantorEntity{
-			Privilege: &milvuspb.PrivilegeEntity{Name: privilegeName},
-			User:      &milvuspb.UserEntity{Name: entity.Grantor.User.Name},
-		})
-	} else {
-		err = proto.Unmarshal([]byte(v), curGrantPrivilegeEntity)
+		var id typeutil.UniqueID
+		id, _, err = kc.idAllocate(1)
```

Comment:
Suggest removing idAllocate method and using a hash value of the key to replacing it.